### PR TITLE
Expose inbounds programmatically

### LINKED ISSUE #55: Store inbounds in a dictionary
So that you could do:

```
tchannel_inbound = rpc.inbounds.get('tchannel')
tchannel = tchannel_inbound.tchannel
```

Larry wants this for the `phone` procedure to dynamically create an outbound based on req params.

### PR DESCRIPTION
**Pull request by breerly**
This allows Larry to easily get at the TChannel instance when dynamically creating an outbound.

Fixes #55

### CODE REVIEW

File: yarpc/rpc.py [hunk 1]
```diff
@@ -50,9 +58,10 @@ def register(self, procedure, handler):
         self._dispatcher.register(procedure, handler)
 
     def start(self):
+
         return [
             gen.maybe_future(inbound.start(self._dispatcher))
-            for inbound in self._inbounds
+            for inbound in self.inbounds.itervalues()
```
Comment (junchaowu) [inline yarpc/rpc.py]: why do you need to make this change?

Comment (breerly) [inline yarpc/rpc.py]: inbounds is now a dictionary

Comment (junchaowu) [inline yarpc/rpc.py]: hmm.. this will bring it different from other language.

File: yarpc/transport/http/inbound.py [hunk 2]
```diff
@@ -33,6 +33,8 @@
 
 class HTTPInbound(object):
 
+    NAME = 'http'
```
Comment (abhinav) [inline yarpc/transport/http/inbound.py]: I'm not sure this is a good idea. This makes it impossible for us to have multiple inbounds of the same type. Off the top of my head, one use case I can think of for that is migration scenarios where you need two HTTP/TChannel servers running as you're consolidating requests made to two different service names into the same service.

**Comment by abhinav** [comment]
Needing access to the underlying TChannel instance is the wrong motivation for this change. That problem can be solved with a different API on the TChannel inbound side rather than changing how RPC manages inbounds in general.

**Comment by breerly** [comment]
Talked offline w @abhinav - going to kill this for now.

**Pull request closed without merging**